src/lib/leaderboard/trpc.ts

```diff
@@ -0,0 +1,27 @@
+
+import { createTRPCRouter, publicProcedure } from "~/lib/trpc/trpc"
+import { orderBy, schema } from "../db"
+import { z } from "zod"
+
+export const leaderboardRouter = createTRPCRouter({
+  getLeaderboard: publicProcedure.input(z.object({
+    limit: z.number().min(1).max(10),
+    cursor: z.number().nullish(),
+  })).query(async ({ ctx, input }) => {
+    const limit = input.limit ?? 10;
+    const { cursor } = input;
+
+    const users = await ctx.db.query.users.findMany({
+      orderBy: [orderBy.desc(schema.users.wins), orderBy.asc(schema.users.name)],
+      limit: limit + 1,
+      offset: cursor ?? 0,
```

Comment:
would be good to not do offset based pagination if we still wanna go with infinite scroll